### PR TITLE
add quantized qkv-fusion pass for x86

### PR DESCRIPTION
**Pull request by yuchengliu1**
This PR add qkv-fusion pass on x86 CPU. It added a manually registered qkv-usion pass. This fusion pass fuse 3 gemm(qkv) and SDPA to a gemm kernel and a SPDA kernel in int8 and fp8.

This PR relies on https://github.com/pytorch/pytorch/pull/184685

### CODE REVIEW

**Review by Valentine233 — COMMENTED** [review]
Thanks!
Could you merge `fp8_concat_linear_pass` into the int8 one, called `quantized_concat_linear`? I suppose most of codes should be the same for two dtypes.

**Comment by mingfeima** [comment]
@claude review this one.

**Comment by Valentine233** [comment]
@claude review code

**Comment by yuchengliu1** [comment]
The bug Claude reported is quite peculiar: int8 and fp8 each have their own execution paths and do not interfere with one another. The remaining suggestions would result in a large amount of defensive code, which is not practical in real scenarios.

File: test/prototype/inductor/test_qsdpa_fusion.py [hunk 1]
```diff
@@ -129,17 +129,17 @@ def forward(self, input):
 class FP8QDQSDPA(torch.nn.Module):
     def __init__(self):
         super().__init__()
-        self.q_out_scale = 1.5
```
Comment (Xia-Weiwen) [inline test/prototype/inductor/test_qsdpa_fusion.py:132]: Why are these changes needed?

File: test/prototype/inductor/test_qsdpa_fusion.py [hunk 2]
```diff
@@ -353,11 +370,30 @@ def _test_int8_sdpa_rewriter(self):
                 num_attention_heads=numhead,
                 attention_head_size=headsize,
             ).eval()
+            mod.init_affine_weights_(
+                [
+                    ("q_proj_weight", 0.0360, 0.0000),
+                    ("k_proj_weight", 0.0320, 0.0000),
+                    ("v_proj_weight", 0.1200, 0.0240),
+                    ("dense_weight", 0.0600, 0.0000),
+                    ("dense_bias", 0.6000, 3.0000),
```
Comment (Xia-Weiwen) [inline test/prototype/inductor/test_qsdpa_fusion.py]: How are these numbers chosen?

File: torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+
+
+def _is_qlinear_target(target):
+    if target in QLINEAR_TARGETS:
+        return True
+    return (
+        callable(target)
+        and getattr(target, "__module__", "")
+        == "torch._inductor.fx_passes.quantization"
+        and getattr(target, "__name__", "") == "qlinear"
```
Comment (Xia-Weiwen) [inline torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py:30]: Better to add a comment about what this is

File: torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+    if dense_values[0].dim() == 0:
+        return torch.stack(dense_values)
+    return torch.cat(dense_values, dim=0)
+
+
+def _is_valid_concat_linear_quantized_fusion(qlinear_nodes, dtype):
+    if len(qlinear_nodes) != 3:
+        return False
+
+    act = _get_node_arg(qlinear_nodes[0], "x", 0)
```
Comment (Xia-Weiwen) [inline torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py:135]: Better to have a comment here to list the arguments of qlinear:
```
qlinear arg list:
0. act
1. act_scale
2. act_zp
3. ...
```

File: test/prototype/inductor/test_qsdpa_fusion.py (hunk 2 above)
Comment (yuchengliu1) [reply test/prototype/inductor/test_qsdpa_fusion.py]: The previous case is too weak to test accuracy. So This PR adjusted the distribution of random numbers and the threshold to enhance the UT. These numbers can scale the result of MHA, match the threshold, and prevent overflow.

File: test/prototype/inductor/test_qsdpa_fusion.py (hunk 1 above)
Comment (yuchengliu1) [reply test/prototype/inductor/test_qsdpa_fusion.py:132]: QKV output scales are merged to one scale in QKV fusion. We need different scales to test the accuracy of this merge process.

File: torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py (hunk 3 above)
Comment (yuchengliu1) [reply torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py:30]: updated

File: torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py (hunk 4 above)
Comment (yuchengliu1) [reply torchao/prototype/inductor/fx_passes/quantized_concat_linear_fusion_cpu.py:135]: comment OP schemas in the head of file

**Review by Valentine233 — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]
